### ISSUE TITLE
I want naming things in code to be considered accessibility

### ISSUE THREAD

**Issue by WebWeWantBot** [WebWeWant/webwewant.fyi#537]
---
title: I want naming things in code to be considered accessibility
date: 2022-03-10T17:25:28.217Z
submitter: askgu.us
number: 622a348855379e0144876df1
tags: [  ]
discussion: https://github.com/WebWeWant/webwewant.fyi/discussions/
status: [ discussing || in-progress || complete ]
related:
  - title:
    url:
    type: [ article || explainer || draft || spec || note || discussion ]
---

Hello Stephanie & co!

My name is Guus "mr Bot" Hoeve, I'm a free-lance ux/dx specialist, focussing my efforts in the front-end where I design like an engineer and develop like an indie, 

I want to raise a point on why naming things in design and/or code should be considered an accessibility issue; it's has a far greater daily impact on the people that actually need to do the work of creating an accessible application.

It's always considered "funny" that there are people that can name things or think that it's something of a profession. Yet most people don't realize that it's one of two most hardest things in CS. Not to my own assesment btw, just google for "naming things is hard".

To open your eyes a bit from my side of things; here's a link to CODELF which shows the most commonly used variable names for... a color; https://unbug.github.io/codelf/#color

Now you see, I'm not interested in a "one-size-fits-all" take on this or just for colors, because I believe insight will lead to it's awareness and/or acceptance first. I'm doing it the Dutch way; straight to the point.

It's just that because naming things is CS is so underrepresented as an accessibility issue for ourselves, and thus the end-user in two-fold if you consider yourself one of them.

Think about it for a second; who does benefit more from a clear and concise naming pattern in their code in the end. I believe a great an accessible UX should therefor start from an equally designed DX to deliver on that promise.

Anyone with a sane mind would instantly recognize that the accessibility issues that the front-end faces, are far more often implied and actually occur than the ones they solve for others. 

That's not to say they are the priority, they are just first and foremost exposed to the disfunctional programming patterns that clog and contextually shift their own minds.

Because it's not just for a scoring-sake that accessibility should drive home it's own point; we're the target audience too you know! And often when you have a great DX, your UX will most certainly benefit from that. Accessibility included if "we" are included too!

That, and business because if you have a '--blue: #000;' that you have to look up in your documentation, which in turn zones you out, then you see the accessibility we need, as in Web We Want, should start with a healthy dose of self-reflection on our own accessability issues.

It's funny though that in the world of ICT, Information and Communication Technology that is build upon the evolution of language from fiction, to form, to function, that in that little world their own inaccessible language itself started to become a problem...

Well I hope that naming things puts that dot on the radar if there are any other witnesess of "the big gap" or the "big devide" between what people can still understand, and what's hidden but should not be forgotten; the basics of language as a technology

Thank you for listening to my properly self-imposed intellectual showpony parade on naming things, but if you have any questions; just askgu.us or followgu.us for any questions on naming things in design systems. 

<hr>

If posted, this will appear at https://webwewant.fyi/wants/622a348855379e0144876df1/
